code/design_pattern/src/builder/builder/Person.java

```diff
@@ -0,0 +1,49 @@
+package builder;
+
+import java.util.Date;
+
+public class Person {
+    private String firstname;
+    private String lastname;
+
+    private Date birthdate;
+    private Nationality nationality;
+    private boolean isProgrammer;
+    private Integer iq;
+    private boolean isPoor;
+    private boolean isPopular;
+    private boolean isEmployed;
+
+    Person(String firstname,
```

Comment:
In your other class, you used the first letter of each parameter for its name, while in this you use the same name as the member variable. Please pick one of the two. Also, I really don't like the extremely large amount of parameters to this constructor.